### PR TITLE
add ability to rerun the job

### PR DESCRIPTION
**Pull request by dawidmalina**
closes https://github.com/erwindon/SaltGUI/issues/141

### CODE REVIEW

**Comment by dawidmalina** [comment]
It is possible to re-run job from jobs page and job detail page.

File: saltgui/static/scripts/routes/route.js [hunk 1]
```diff
@@ -39,4 +39,62 @@ class Route {
...
+    }
+
+    target.value = targetString;
+    command.value = commandString;
+    // the menu may become (in)visible due to content of command field
+    this.router.commandbox.cmdmenu.verifyAll();
+  }
+
+  _decodeArgumentsText(rawArguments) {
+    console.log(rawArguments);
```
Comment (erwindon) [inline saltgui/static/scripts/routes/route.js]: please remove this one

File: saltgui/static/scripts/routes/route.js [hunk 2]
```diff
@@ -39,4 +39,62 @@ class Route {
...
+    const rawObject = rawArguments;
+    let argumentsText = "";
+
+    switch (typeof rawObject) {
+    case "undefined":
+      argumentsText = "";
+      break;
+    case "object": {
```
Comment (erwindon) [inline saltgui/static/scripts/routes/route.js]: does anything bad happen without the { ... } ?

File: saltgui/static/scripts/routes/page.js [hunk 3]
```diff
@@ -335,6 +336,9 @@ class PageRoute extends Route {
     menu.addMenuItem("Show&nbsp;details", function(evt) {
       window.location.assign("/job?id=" + encodeURIComponent(job.id));
     }.bind(this));
+    menu.addMenuItem("Re-run&nbsp;job...", function(evt) {
+      this._runFullCommand(evt, job["Target-type"], job.Target, functionText);
```
Comment (erwindon) [inline saltgui/static/scripts/routes/page.js]: can you intercept the value "unknown-target".
In that case, Target-type should be default and Target should be empty.
Note that the command parameters are also lost in that case

File: saltgui/static/scripts/routes/route.js (hunk 2 above)
Comment (dawidmalina) [reply saltgui/static/scripts/routes/route.js]: Yes. I don't remember exact message but eslint was failing without it.

File: saltgui/static/scripts/routes/page.js (hunk 3 above)
Comment (dawidmalina) [reply saltgui/static/scripts/routes/page.js]: Ok I will try to handle "unknown-target" case and use default values it such situation.

**Comment by dawidmalina** [comment]
For the remaining types of re-run (which you described) I propose to do it with dedicated pull request.

**Pull request merged**